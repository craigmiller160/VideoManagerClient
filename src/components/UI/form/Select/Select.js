import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';
import { FormGroup, Label } from 'reactstrap';
import newid from '../../../../utils/newid';
import formStyles from '../FormStyles.scss';

const Select = (props) => {
    const {
        label,
        name,
        options,
        onChange,
        value
    } = props;

    const onChangeFn = onChange ? (value) => {
        onChange({name, value: value.value});
    } : null;

    const id = newid();
    return (
        <FormGroup className={ formStyles['input-label'] }>
            <Label
                for={ id }
            >
                { label }
            </Label>
            <ReactSelect
                id={ id }
                name={ name }
                options={ options }
                onChange={ onChangeFn }
                isSearchable={ true }
                value={ value }
            />
        </FormGroup>
    );
};

Select.propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    name: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.string
};

export default Select;