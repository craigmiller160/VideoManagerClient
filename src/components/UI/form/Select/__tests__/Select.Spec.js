import React from 'react';
import Select from '../Select';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Select', () => {
    const props = {
        label: 'My Select',
        name: 'MySelect',
        options: [
            { label: 'Opt1', value: 'opt1' },
            { label: 'Opt2', value: 'opt2' }
        ]
    };

    it('should render correctly', () => {
        const component = shallow(
            <Select { ...props } />
        );
        expect(component).toMatchSnapshot();
    });

    it('should handle onChange', () => {
        let value = 'InitialValue';
        props.onChange = (newValue) => value = newValue;

        const component = shallow(
            <Select { ...props } />
        );

        const result = component.find('[name="MySelect"]');
        expect(result.length).toEqual(1);

        result.simulate('change', { label: 'Opt2', value: 'opt2' });
        expect(value).toEqual({ label: 'Opt2', value: 'opt2' });
    });
});