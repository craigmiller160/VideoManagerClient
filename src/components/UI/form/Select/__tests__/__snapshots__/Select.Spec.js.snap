// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    label="My Select"
    name="MySelect"
    options={
      Array [
        Object {
          "label": "Opt1",
          "value": "opt1",
        },
        Object {
          "label": "Opt2",
          "value": "opt2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          for="id_0"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          My Select
        </Label>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="id_0"
          isSearchable={true}
          name="MySelect"
          onChange={null}
          options={
            Array [
              Object {
                "label": "Opt1",
                "value": "opt1",
              },
              Object {
                "label": "Opt2",
                "value": "opt2",
              },
            ]
          }
        />,
      ],
      "className": undefined,
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "My Select",
          "for": "id_0",
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": "My Select",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "id": "id_0",
          "isSearchable": true,
          "name": "MySelect",
          "onChange": null,
          "options": Array [
            Object {
              "label": "Opt1",
              "value": "opt1",
            },
            Object {
              "label": "Opt2",
              "value": "opt2",
            },
          ],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            for="id_0"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            My Select
          </Label>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            id="id_0"
            isSearchable={true}
            name="MySelect"
            onChange={null}
            options={
              Array [
                Object {
                  "label": "Opt1",
                  "value": "opt1",
                },
                Object {
                  "label": "Opt2",
                  "value": "opt2",
                },
              ]
            }
          />,
        ],
        "className": undefined,
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "My Select",
            "for": "id_0",
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": "My Select",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "id": "id_0",
            "isSearchable": true,
            "name": "MySelect",
            "onChange": null,
            "options": Array [
              Object {
                "label": "Opt1",
                "value": "opt1",
              },
              Object {
                "label": "Opt2",
                "value": "opt2",
              },
            ],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
